### ISSUE TITLE
RFE: open PDF in an external browser

### ISSUE THREAD

**Issue by ToddAndMargo** [brave/brave-browser#8590]
Request for Enhancement: 

May we please have the option to open a PDF in an external browser, as does Firefox and Vivaldi?

Many thanks,
-T

**Comment by rebron** [comment]
Hi @ToddAndMargo 

It's a bit of a buried setting but you can toggle this setting and you'll be able to open PDFs in an external pdf viewer. 

brave://settings/content/pdfDocuments

**Comment by ToddAndMargo** [comment]
On 2020-03-09 09:24, Rafael Ebron wrote:
> Hi @ToddAndMargo <https://github.com/ToddAndMargo>
> 
> It's a bit of a buried setting but you can toggle this setting and 
> you'll be able to open PDFs in an external pdf viewer.
> 
> brave://settings/content/pdfDocuments

Hi Rafael,

I wrote it down to carry with me to customer sites.
Thank you!

Any chance of making it easier to find?

-T

**Comment by ToddAndMargo** [comment]
Hi Rafael,

That is a nice setting if yo always want to save a pdf to disk.

It is also not what I asked for.  I want the open to open the PDF in an external viewer from Brave, as does Firefox and Vivaldi, not just download the file.

Please reopen this request

**Comment by rebron** [comment]
@ToddAndMargo I understand the request, choose application for file handling of .pdf files.

**Comment by ToddAndMargo** [comment]
On 2020-03-16 16:36, Rafael Ebron wrote:
> @ToddAndMargo <https://github.com/ToddAndMargo> I understand the 
> request, choose application for file handling of .pdf files.

Hi Rafael,

I am not sure I am understand.  Did I miss a function
that is already there, again, or are you placing the request
to add the feature?

Many thanks,
-T

**Comment by rebron** [comment]
@ToddAndMargo This particular feature isn't there, I'm adding some more clairity to the feature request. It's basically protocol and file handling so for mailto: open x application, for .pdf open pdf viewer or acrobat, etc.

It's related to sites asking to be handlers for things like mailto: or calendar links. Firefox has better ui for it, it's an old feature.

e.g. brave://settings/handlers
<img width="701" alt="Screen Shot 2020-03-16 at 6 34 07 PM" src="https://user-images.githubusercontent.com/5951041/76813298-bf54c780-67b4-11ea-852e-4c21f0cb78ff.png">

**Comment by ToddAndMargo** [comment]
Perfect!

**Comment by federerfanatic** [comment]
I am currently using Version 1.7.98 Chromium: 81.0.4044.113 (Official Build) (64-bit), on
lsb_release -a
LSB Version:	core-11.0.1ubuntu1-noarch:security-11.0.1ubuntu1-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 19.10
Release:	19.10
Codename:	eoan


BRAVE has suddenly refused to run the system pdf viewer. It always opens it in BRAVE.
I currently have brave://settings/content/pdfDocuments?search=pdf to Download file. I also
allow myself to choose save location. I also set mimeopen --ask on a pdf file and told it to use okular which I regard as my system viewer

**Comment by EmyrMacdonald** [comment]
I download 20-30 pdf files per day, opening and editing each.  I need to be able to open pdf's directly with an external viewer/editor rather than opening within the browser.  Currently, I use Firefox for this reason alone.  Brave is extremely attractive as a browser, but until there is a capability to open pdf's with an external viewer, it won't be my default browser.

**Comment by ToddAndMargo** [comment]
Hi Emyr Macdonald,

If you want to use a Blink based browser with this feature, Vivaldi has it.  I have had to switch several customers to Vivaldi over this very issue.  I would have used Firefox, but too many of their business to business portals no longer work with Firefox.  Vivaldi does take a little getting use to though.

-T

**Comment by EmyrMacdonald** [comment]
Many thanks Todd. Best, EmyrSent from my Samsung Galaxy smartphone.
-------- Original message --------From: ToddAndMargo <notifications@github.com> Date: 16/05/2020  23:14  (GMT+00:00) To: brave/brave-browser <brave-browser@noreply.github.com> Cc: EmyrMacdonald <emyr.macdonald@gmail.com>, Comment <comment@noreply.github.com> Subject: Re: [brave/brave-browser] RFE: provide option to choose an application when handling .pdf files (#8590) 
Hi Emyr Macdonald,
If you want to use a Blink based browser with this feature, Vivaldi has it.  I have had to switch several customers to Vivaldi over this very issue.  I would have used Firefox, but too many of their business to business portals no longer work with Firefox.  Vivaldi does take a little getting use to though.
-T

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/brave/brave-browser/issues/8590#issuecomment-629712794",
"url": "https://github.com/brave/brave-browser/issues/8590#issuecomment-629712794",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by ToddAndMargo** [comment]
Hi Guys,

This has become a **YUGE issue** with my customers.  Would you please jump on it as soon as possible!  It is very difficult to keep them off of Chrome, etc. after they find this ability is missing in Brave Browser.  Would you please put this at the top of your list!

Many thanks,
-T

**Comment by ToddAndMargo** [comment]
I have a customer who can not log into DropBox with Chrome.  I can't figure it out -- it works on every other browser (a complete Chrome reset did not even work).  Firefox does not work properly on too many of here web sites, so Blink Engine it is.  So I put her on Brave, where DropBox works perfectly.  And she instantly tripped across this defect.   Now she is on the absolute war path over it.  I can put her on Vivaldi, which does not have this defect, but Vivaldi is too intensely complicated for her.  Please, please hurry up and fix this!!!!

**Comment by ToddAndMargo** [comment]
Several customer back is dealing with a government procurement portal with a lot of PDF's.  He need to use an external view as the PDF are badly formed and difficult to view properly in a browser.  I had to switch to Vivaldi over this issue.  This is what he said to me "When I [expletive] click on a [expletive] PDF, I what the [expletive] to [expletive] open up  and when I click [expletive] print, I want it to [expletive] print."  He was pretty mad to say the least.  He would not put up with the download and open method in Brave.  To customers, they DO NOT CARE if things are done differently, in Brave.  They only want what they expect the behavior to be.  And they really, really do not appreciate extra steps that they refuse to learn.

**Comment by ToddAndMargo** [comment]
I just got off a Go To Assist session with a customer that needs to fill out a government certification every week.  It won't function properly in Firefox, but does in Blink.  The end game is a PDF, but it won't display properly in Brave, so I switched her to download rather than open.  It displays properly in Acrobat Reader.  She said she understood how, but her voice was confused.  So ...  (At least there was no cussing this time.)

**Comment by ToddAndMargo** [comment]
Hi Guys,

The lack of this feature is DRIVING MY CUSTOMERS CRAZY and off Brave.  Would some kind hearted developer please implement this as soon as possible?

Many thanks,
-T

**Comment by EmyrMacdonald** [comment]
I’m another who has left Brave because of its inability to open pdf files in an external viewer.  For any research use, this is imperative.

Regards

Emyr

**Comment by MEKadan** [comment]
It still isn't implemented!?!
 Because G**gle is so f*cking sh*t these days, could not find inf on the features I was missing before installation, I had to just install & try. Those features on top of list being opening pdf in external reader and multi-row tabs.

Anyway, have fun. Goodbye & uninstalling (again)

**Comment by dhunter49** [comment]
I agree with the OP. Having a way to open PDF files in Okular or any other desktop app would be great. I don't think a *libre* app should decide on it's own how to open certain files. I don't think it's necessary to provide a list to pick apps from, but just an option to let OS handle it.
